lib/superstore/types/range_type.rb

```diff
@@ -0,0 +1,45 @@
+module Superstore
+  module Types
+    class RangeType < BaseType
+      class_attribute :subtype
+
+      def encode(range_hash)
+        [
+          convert(:encode, range_hash[:lower]),
+          convert(:encode, range_hash[:upper])
+        ]
+      end
+
+      def decode(range_tuple)
+        {
+          lower: convert(:decode, range_tuple[0]),
+          upper: convert(:decode, range_tuple[1])
+        }
+      end
```

Comment:
This doesn't make sense to me.  `decode` always accepts a string.  Typecast is used for non-string instantiation.